src/privacy_accounting/optimizer/dp_optimizer.py

```diff
@@ -53,13 +53,17 @@ def process_microbatch(losses, sample_state):
         for losses in microbatches_losses:
             sample_state = process_microbatch(losses, sample_state)  # accumulate up the clipped microbatch gradients
 
-        final_grads, new_global_state = self.dp_sum_query.get_noised_result(sample_state, self._global_state)
-        self._global_state = new_global_state
+        final_grads, _ = self.dp_sum_query.get_noised_result(sample_state, self._global_parameters)
 
         self.apply_grads(param_groups, grads=final_grads)
 
-        self.optimiser.step()
-        self.optimiser.zero_grad()
+        self.optimizer.step()
+        self.optimizer.zero_grad()
+
+        # compute the new loss
+        current_loss = torch.sum(self.loss_per_example(self.model(x), y))
```

Comment:
does this quantity not exist already? Just sum it from the original loss call rather than calling the model twice

Reply:
the model parameters have changed - granted though this is slightly inefficient.